libs/hiraya/src/components/Button/style.ts

```diff
@@ -0,0 +1,29 @@
+import { Button as MUIButton, styled } from '@mui/material';
+
+import { variants, buttonSizesStyle, buttonWidthStyle } from './theme';
+
+interface ButtonProps {
+  label?: string;
+  variantStyle?: 'primary' | 'secondary' | 'tertiary';
+  buttonSizes?: 'small' | 'medium' | 'large';
+  buttonWidthSizes?: 'isFullWidth' | 'iconOnly' | 'default';
+}
+
+const ButtonStyled = styled(MUIButton)<ButtonProps>(
+  ({ variantStyle, buttonSizes, buttonWidthSizes }) => ({
+    backgroundColor: variants[variantStyle || 'primary'].background,
+    color: variants[variantStyle || 'primary'].text,
+    padding: '10px 20px',
+    '&:height': buttonSizesStyle[buttonSizes || 'small'].height,
+    '&:width': buttonWidthStyle[buttonWidthSizes || 'default'],
```

Comment:
Also could you explain to me what "&:height" does? this is the first time i've seen it,

PS: your file structure looks good!